src/CommunityToolkit.Maui/ImageSources/GravatarImageSource.shared.cs

```diff
@@ -0,0 +1,239 @@
+
+	/// <summary>The backing store for the <see cref="ParentWidth" /> bindable property.</summary>
+	internal static readonly BindableProperty ParentWidthProperty = BindableProperty.Create(nameof(ParentWidth), typeof(int), typeof(GravatarImageSource), defaultValue: defaultSize, propertyChanged: OnSizePropertyChanged);
+
+	const int cancellationTokenSourceTimeout = 737;
+	const string defaultGravatarImageAddress = "https://www.gravatar.com/avatar/";
+	const int defaultSize = 80;
+	static readonly HttpClient singletonHttpClient = new();
+	int gravatarSize;
+	CancellationTokenSource? tokenSource;
+
+	/// <summary>Initializes a new instance of the <see cref="GravatarImageSource"/> class.</summary>
+	public GravatarImageSource()
+	{
+		Stream = new Func<CancellationToken, Task<Stream>>(cancelationToken => DownloadStreamAsync(Uri, cancelationToken));
+		Uri = new Uri(defaultGravatarImageAddress);
+	}
+
+	/// <summary>Gets or sets a <see cref="TimeSpan"/> structure that indicates the length of time after which the image cache becomes invalid.</summary>
+	public TimeSpan CacheValidity
+	{
+		get => (TimeSpan)GetValue(CacheValidityProperty);
+		set => SetValue(CacheValidityProperty, value);
+	}
+
+	/// <summary>Gets or sets a Boolean value that indicates whether caching is enabled on this <see cref="GravatarImageSource"/> object.</summary>
+	public bool CachingEnabled
+	{
+		get => (bool)GetValue(CachingEnabledProperty);
+		set => SetValue(CachingEnabledProperty, value);
+	}
+
+	/// <summary>Gets or sets the email address.</summary>
+	public string? Email
+	{
+		get => (string)GetValue(EmailProperty);
+		set => SetValue(EmailProperty, value);
+	}
+
+	/// <summary>Gets or sets the default image.</summary>
+	public DefaultImage Image
+	{
+		get => (DefaultImage)GetValue(ImageProperty);
+		set => SetValue(ImageProperty, value);
+	}
+
+	/// <summary>Gets a value indicating whether the control email is empty.</summary>
+	public override bool IsEmpty => string.IsNullOrEmpty(Email);
+
+	/// <summary>Gets or sets the URI for the image to get.</summary>
+	[System.ComponentModel.TypeConverter(typeof(UriTypeConverter))]
+	public Uri Uri { get; set; }
+
+	/// <summary>Gets or sets the parent height.</summary>
+	internal int ParentHeight => (int)GetValue(ParentHeightProperty);
+
+	/// <summary>Gets or sets the parent width.</summary>
+	internal int ParentWidth => (int)GetValue(ParentWidthProperty);
+
+	/// <summary>Gets or sets the image size.</summary>
+	int GravatarSize
+	{
+		get => gravatarSize;
+		set => gravatarSize = Math.Clamp(value, 1, 2048);
+	}
+
+	/// <summary>Returns the Uri as a string.</summary>
+	/// <returns>String of the URI.</returns>
+	public override string ToString()
+	{
+		return $"Uri: {Uri}, Email:{Email}, Size:{GravatarSize}, Image: {DefaultGravatarName(Image)}, CacheValidity:{CacheValidity}, CachingEnabled:{CachingEnabled}";
+	}
+
+	/// <summary>On parent set.</summary>
+	protected override void OnParentSet()
+	{
+		object element = Parent;
```

Comment:
this variable can be removed. use Parent directly in if

Reply:
Done.